node/pkg/websocketfetcher/providers/kucoin/utils.go

```diff
@@ -1,35 +1,24 @@
 package kucoin
 
 import (
-	"fmt"
-	"strings"
 	"time"
 
 	"bisonai.com/orakl/node/pkg/websocketfetcher/common"
 )
 
-func RawDataToFeedData(raw Raw, feedMap map[string]int32) (*common.FeedData, error) {
-	feedData := new(common.FeedData)
+func RawDataToFeedData(raw SymbolSnapshotRaw, feedMap map[string]int32) *common.FeedData {
+	snapshot := raw.Data.Data
+	symbol := snapshot.Symbol
+	id := feedMap[symbol]
 
-	timestamp := time.Unix(raw.Data.Time/1000, 0)
-	value, err := common.PriceStringToFloat64(raw.Data.Price)
-	if err != nil {
-		return feedData, err
-	}
-	rawPair := strings.TrimPrefix(raw.Topic, "/market/ticker:")
-	splitted := strings.Split(rawPair, "-")
-	if len(splitted) < 2 {
-		return feedData, fmt.Errorf("invalid feed name")
-	}
-	target := splitted[0]
-	quote := splitted[1]
+	timestamp := time.Unix(snapshot.Time/1000, 0)
```

Comment:
@nick-bisonai Just suggestion

```suggestion
	timestamp := time.UnixMilli(snapshot.Time, 0)
```

Reply:
There are multiple places with this pattern, I'll update all occurences